src/parameterizations/vertical/MOM_energetic_PBL.F90

```diff
@@ -1916,13 +1759,184 @@ subroutine find_PE_chg_orig(Kddt_h, h_k, b_den_1, dTe_term, dSe_term, &
+      ! The limit for rotation (Ekman length) limited mixing
+      mstar_N = 0.0
+      if (Ustar > Abs_Coriolis * BLD) mstar_N = CS%C_EK * log(Ustar / (Abs_Coriolis * BLD))
+    endif
+
+    ! Here 1.25 is about .5/von Karman, which gives the Obukhov limit.
+    MStar = max(MStar_S, min(1.25, MStar_N))
+    if (CS%MStar_Cap > 0.0) MStar = min( CS%MStar_Cap,MStar )
+  elseif ( CS%mstar_scheme == MStar_from_RH18 ) then
+    if (CS%answers_2018) then
+      MStar_N = CS%RH18_MStar_cn1 * ( 1.0 - 1.0 / ( 1. + CS%RH18_MStar_cn2 * &
+                exp( CS%RH18_mstar_CN3 * BLD * Abs_Coriolis / UStar) ) )
+    else
+      MSN_term = CS%RH18_MStar_cn2 * exp( CS%RH18_mstar_CN3 * BLD * Abs_Coriolis / Ustar)
+      MStar_N = (CS%RH18_MStar_cn1 *  MSN_term) / ( 1. + MSN_term)
+    endif
+    MStar_S = CS%RH18_MStar_CS1 * ( max(0.0, Buoyancy_Flux)**2 * BLD / &
+             ( Ustar**5 * max(Abs_Coriolis,1.e-20*US%T_to_s) ) )**CS%RH18_mstar_cs2
+    MStar = MStar_N + MStar_S
+  endif
+
+  !/ 2. Adjust mstar to account for convective turbulence
+  if (CS%answers_2018) then
+    MStar_Conv_Red = 1. - CS%MStar_Convect_coef * (-min(0.0,Buoyancy_Flux) + 1.e-10*US%T_to_s**3*US%m_to_Z**2) / &
+                         ( (-min(0.0,Buoyancy_Flux) + 1.e-10*US%T_to_s**3*US%m_to_Z**2) + &
+                         2.0 *MStar * Ustar**3 / BLD )
+  else
+    MSCR_term1 = -BLD * min(0.0, Buoyancy_Flux)
+    MSCR_term2 = 2.0*MStar * Ustar**3
+    MStar_Conv_Red = ((1.-CS%mstar_convect_coef) * MSCR_term1 + MSCR_term2) / (MSCR_term1 + MSCR_term2)
+  endif
+
+  !/3. Combine various mstar terms to get final value
+  MStar = MStar * MStar_Conv_Red
+
+  if (present(Langmuir_Number)) then
+    !### In this call, ustar was previously ustar_mean.  Is this change deliberate?
+    call mstar_Langmuir(CS, US, abs_Coriolis, buoyancy_flux, ustar, BLD, Langmuir_number, mstar, &
+                        mstar_LT, Convect_Langmuir_Number)
+  endif
+
+end subroutine Find_Mstar
+
+!> This subroutine modifies the Mstar value if the Langmuir number is present
+subroutine Mstar_Langmuir(CS, US, abs_Coriolis, buoyancy_flux, ustar, BLD, Langmuir_Number, &
+                          mstar, mstar_LT, Convect_Langmuir_Number)
+  type(energetic_PBL_CS), pointer    :: CS    !< Energetic_PBL control structure.
+  type(unit_scale_type), intent(in)  :: US    !< A dimensional unit scaling type
+  real,                  intent(in)  :: Abs_Coriolis !< Absolute value of the Coriolis parameter [T-1 ~> s-1]
+  real,                  intent(in)  :: Buoyancy_Flux !< Buoyancy flux [Z2 T-3 ~> m2 s-3]
+  real,                  intent(in)  :: UStar !< Surface friction velocity with? gustiness [Z T-1 ~> m s-1]
+  real,                  intent(in)  :: BLD   !< boundary layer depth [Z ~> m]
+  real,                  intent(inout) :: Mstar !< Input/output mstar (Mixing/ustar**3) [nondim]
+  real,                  intent(in)  :: Langmuir_Number !Langmuir number [nondim]
```

Comment:
This line has incorrect syntax for doxygen markup: the "!" should be "!< " (including the white space please).

Reply:
The requested modification has been changed, and the revised pull request is now passing all checks.